cheroot/server.py

```diff
@@ -340,8 +341,65 @@ def __next__(self):
 
     next = __next__
 
+class RFile:
+    def __init__(self, rfile):
+        self.rfile = rfile
+
+    def read(self, size=None):
+        raise NotImplementedError
+
+    def readline(self, size=None):
+        raise NotImplementedError
+
+    def readlines(self, sizehint=None):
+        raise NotImplementedError
+
+    def close(self):
+        self.rfile.close()
+
+class HyperRFile(RFile):
+
+    def __init__(self, rfile, conn):
+        super(HyperRFile, self).__init__(rfile)
+        self.conn = conn
+
+    def read(self, size=None):
+        if self.conn.their_state != h11.SEND_BODY or size == 0:
+            return b""
+        event = self.conn.next_event()
+        if event is h11.NEED_DATA:
+            self.conn.receive_data(self.rfile.read(size))
+            event = self.conn.next_event()
+        assert isinstance(event, h11.Data)
+        return event.data
 
-class KnownLengthRFile:
+    def readline(self, size=None):
+        if self.conn.their_state != h11.SEND_BODY or size == 0:
+            return b""
+        event = self.conn.next_event()
+        if event is h11.NEED_DATA:
+            self.conn.receive_data(self.rfile.readline(size))
+            event = self.conn.next_event()
+        assert isinstance(event, h11.Data)
+        return event.data
+
+    def readlines(self, sizehint=None):
+        if self.conn.their_state != h11.SEND_BODY:
+            return b""
+        lines = []
+        line = self.rfile.readline()
+        while line:
+            event = self.conn.next_event()
+            if event is h11.NEED_DATA:
```

Comment:
plz reverse this check in order to reduce the following block indentation
```suggestion
            if event is not h11.NEED_DATA:
                continue
```